pkg/apis/core/v1/defaults.go

```diff
@@ -203,6 +205,11 @@ func SetDefaults_PodSpec(obj *v1.PodSpec) {
 	if obj.SchedulerName == "" {
 		obj.SchedulerName = v1.DefaultSchedulerName
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.WorkloadInfoDefaulting) {
```

Comment:
Continue on the discussion if the feature gate is needed as in the PR conversation here:
my understanding for fixing issue #698 , this feature-gate will need to be set true anyways, so from that perspective, this feature gate ( at least per the code here ) is not much useful.  if there are other code path in Arktos that will need this flag to be false, i think we should consider to fix that code path, vm and container support as the unified platform is essential in Arktos, i.e. from that perspective, it is not just a "feature", right ?

Reply:
Yes. I'm setting it to true in arktos-up and kube-up to cover our deployment use cases. This is apiserver defaulting code, and there is no other place where this is required. Once we initialize this, the deserialization of PodSpec should have WorkloadInfo allocated and set. If we were to do this without featuregate, it will require investigating and changing to the test infra to consider this special-case, and fixing all the failing unit tests.